tests/framework/project/test_pipeline_discovery.py

```diff
@@ -83,6 +83,37 @@ def test_find_pipelines_skips_modules_without_create_pipelines_function(
     assert sum(pipelines.values()).outputs() == pipeline_names
 
 
+@pytest.mark.parametrize(
+    "mock_package_name_with_pipelines,pipeline_names",
+    [(x, x) for x in [set(), {"my_pipeline"}]],
+    indirect=True,
+)
+def test_find_pipelines_skips_hidden_modules(
+    mock_package_name_with_pipelines, pipeline_names
+):
+    pipelines_dir = Path(sys.path[0]) / mock_package_name_with_pipelines / "pipelines"
+    pipeline_dir = pipelines_dir / ".hidden"
```

Comment:
```suggestion
    pipeline_dir = pipelines_dir / ".ipynb_checkpoints"
```

I think it might be nice to make this realistic so people reading this in 2 years time realise why it's in there!